pkg/apis/kudo/v1beta1/instance_admission_test.go

```diff
@@ -0,0 +1,363 @@
+package v1beta1
+
+import (
+	"log"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kudobuilder/kudo/pkg/util/kudo"
+)
+
+func TestValidateUpdate(t *testing.T) {
+	// because Go doesn't let us take a pointer of a literal
+	deploy := "deploy"
+	update := "update"
+
+	ov := &OperatorVersion{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo-operator", Namespace: "default"},
+		TypeMeta:   metav1.TypeMeta{Kind: "OperatorVersion", APIVersion: "kudo.dev/v1beta1"},
+		Spec: OperatorVersionSpec{
+			Plans: map[string]Plan{"deploy": {}, "update": {}},
+			Parameters: []Parameter{
+				{
+					Name:    "foo",
+					Trigger: deploy,
+				},
+				{
+					Name:    "other-foo",
+					Trigger: deploy,
+				},
+				{
+					Name:    "bar",
+					Trigger: update,
+				},
+			},
+		},
+	}
+
+	idle := &Instance{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "kudo.dev/v1beta1",
+			Kind:       "Instance",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "test",
+		},
+		Spec: InstanceSpec{
+			OperatorVersion: v1.ObjectReference{
+				Name: "foo-operator",
+			},
+			Parameters: map[string]string{
+				"foo": "foo",
+			},
+		},
+		Status: InstanceStatus{},
+	}
+
+	scheduled := idle.DeepCopy()
+	scheduled.Spec.PlanExecution = PlanExecution{PlanName: deploy}
+
+	upgraded := idle.DeepCopy()
+	upgraded.Spec.OperatorVersion = v1.ObjectReference{Name: "foo-operator-2.0"}
+
+	tests := []struct {
```

Comment:
These tests cover all the rules defined in the `instance_admission.go` webhook.

Reply:
nice tests 👏

Reply:
🙇